clang/test/SemaSYCL/max-concurrency.cpp

```diff
@@ -0,0 +1,125 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -sycl-std=2020  -fsyntax-only -ast-dump -verify -pedantic %s | FileCheck %s
+
+#include "sycl.hpp"
+
+using namespace cl::sycl;
+queue q;
+
+class Functor1 {
+public:
+  [[intel::max_concurrency(4)]] void operator()() const {}
+};
+
+[[intel::max_concurrency]] void foo() {} // expected-error {{'max_concurrency' attribute takes one argument}}
+
+class Functor2 {
+public:
+  void operator()() const {
+    foo();
+  }
+};
+
+template <int NT>
+class Functor3 {
+public:
+  [[intel::max_concurrency(NT)]] void operator()() const {}
+  // expected-error@+1 {{'max_concurrency' attribute only applies to 'for', 'while', 'do' statements, and (device) functions}}
+  [[intel::max_concurrency(2)]] int a[10];
+};
+
+// expected-error@+1 {{'max_concurrency' attribute takes one argument}}
+[[intel::max_concurrency(3, 3)]] void goo() {}
+
+class Functor4 {
+public:
+  void operator() () const {
+    goo();
+  }
+};
+
+// expected-error@+1 {{'max_concurrency' attribute requires a positive integral compile time constant expression}}
+[[intel::max_concurrency(-1)]] void bar() {}
+class Functor5 {
+public:
+  void operator() () const {
+    bar();
+  }
+};
+
+// expected-error@+1 {{integral constant expression must have integral or unscoped enumeration type, not 'const char [16]'}}
+[[intel::max_concurrency("numberofthreads")]] void zoo() {}
+
+template <int NT>
+[[intel::max_concurrency(NT)]] void func() {}
+
+[[intel::max_concurrency(8)]] void dup();
+[[intel::max_concurrency(9)]] void dup() {} // expected-error {{duplicate Intel FPGA function attribute 'max_concurrency'}}
+
```

Comment:
Could you also add an example where the declaration and definition use the same value?